### PR TITLE
SONARJAVA-4857: FP in S3457 on certain java.util.logging strings with single quotes

### PR DESCRIPTION
**Pull request by ADarko22**
…ngle quotes

Please ensure your pull request adheres to the following guidelines: 

- [ ] Use the following formatting style: [SonarSource/sonar-developer-toolset](https://github.com/SonarSource/sonar-developer-toolset#code-style)
- [ ] Unit tests are passing and you provided a unit test for your fix
- [ ] ITs should pass : To run ITs locally, checkout the [README](https://github.com/SonarSource/sonar-java/blob/master/README.md) of the project.
- [ ] If there is a [Jira](http://jira.sonarsource.com/browse/SONARJAVA) ticket available, please make your commits and pull request start with the ticket number (SONARJAVA-XXXX)

### CODE REVIEW

File: java-checks-test-sources/default/src/main/java/checks/PrintfMisuseCheck.java [hunk 1]
```diff
@@ -254,6 +254,10 @@ void foo(Calendar c) throws IOException {
     logger4.log(level, "message " + "...");
     logger4.log(level, "message " + param1, new Exception()); // Noncompliant {{Lambda should be used to defer string concatenation.}}
 
+    logger4.log(level, "Can't load library \"{0}\"!", "foo"); // Noncompliant {{Single quote "'" must be escaped.}}
```
Comment (kaufco) [inline java-checks-test-sources/default/src/main/java/checks/PrintfMisuseCheck.java:257]: The message says "Single quote "'" must be escaped.". How is it supposed to be escaped? `Can\'t` or `Can''t`? 
- `Can\'t`: it still raises an issue. If this is the supposed escape, then the rule must include an exception for it.
- `Can''t`: I'm not sure it produces the expected output "Can't", we should make this sure by checking the docs or test it ourselves if not documented explicitly.

**Review by kaufco — CHANGES_REQUESTED** [review]
Looks good, except for one question about how to escape singular quotes. See the comment inline.

**Review by kaufco — APPROVED** [review]
Lgtm!

**Comment by ADarko22** [comment]
Thank you @kaufco, I've updated the rspec to mention the double single quote for escaping a single quote: https://github.com/SonarSource/rspec/pull/3753. It should be easier for the users now :)

**Pull request merged**